include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,575 @@
+
+	/** Dynamic Huffman code */
+	ODP_COMP_HUFFMAN_DYNAMIC,
+
+	/** Default huffman code selected by implementation */
+	ODP_COMP_HUFFMAN_DEFAULT,
+} odp_comp_huffman_code_t;
+
+/**
+ * Compression DEFLATEe algorithm parameters.
+ * Also initialized by other deflate based algorithms , ex. ZLIB
+ */
+typedef struct odp_comp_deflate_param {
+	/**
+	 * Compression level
+	 *
+	 * Valid range is integer between (0 ... max_level)
+	 * level supported by the implementation.
+	 *
+	 * where,
+	 * 0 - implemention default
+	 *
+ 	 * 1 - fastest compression i.e. output produced at
+ 	 * best possible speed at the expense of compression quality
+	 *
+	 * max_level - High quality compression
+	 *
+	 * @see 'max_level' in odp_comp_alg_capability_t
+	 */
+	uint32_t comp_level;
+
+	/** huffman code to use */
+	odp_comp_huffman_code_t huffman_code;
+} odp_comp_deflate_param_t;
+
+/**
+ * Compression algorithm specific parameters
+ */
+typedef union odp_comp_alg_param_t {
+	/** deflate parameter */
+	odp_comp_deflate_param_t deflate;
+
+	/**
+	 * Struct for defining zlib algorithm parameters
+	 */
+	struct comp_alg_zlib_param {
+		/** deflate algo params */
+		odp_comp_deflate_param_t deflate;
+	} zlib; /** zlib parameter */
+} odp_comp_alg_param_t;
+
+ /**
+ * Compression session creation parameters
+ */
+typedef struct odp_comp_session_param_t {
+	/** Compression operation type Compress vs Decompress */
+	odp_comp_op_t op;
+
+	/** Compression operation mode
+	 *
+	 * Operation mode Synchronous vs Asynchronous
+	 *
+	 * @see odp_comp_op(), odp_comp_op_enq()
+	 */
+	odp_comp_op_mode_t mode;
+
+	/** Compression algorithm
+	 *
+	 *  @see odp_comp_capability()
+	 */
+	odp_comp_alg_t comp_algo;
+
+	/** Hash algorithm
+	 *
+	 *  @see odp_comp_alg_capability()
+	 */
+	odp_comp_hash_alg_t hash_algo;
+
+	/** parameters specific to compression */
+	odp_comp_alg_param_t alg_param;
+
+	/** Session packet enqueue ordering
+	 * Boolean to indicate if packet enqueue ordering is required per
+	 * session. Valid only for Asynchronous operation mode
+	 * (ODP_COMP_OP_MODE_ASYNC). Packet order is always maintained for
+	 * synchronous operation mode (ODP_COMP_OP_MODE_SYNC)
+	 *
+	 * true: packet session enqueue order maintained
+	 *
+	 * false: packet session enqueue order is not maintained
+	 *
+	 * @note: By disabling packet order requirement, performance oriented
+	 * application can leverage HW offered parallelism to increase operation
+	 * performance.
+	 */
+	odp_bool_t packet_order;
+
+	/** Destination queue for compression operations result.
+	 * Results are enqueued as ODP_EVENT_PACKET with subtype
+	 * ODP_EVENT_PACKET_COMP
+	 */
+	odp_queue_t compl_queue;
+} odp_comp_session_param_t;
+
+/**
+ * Compression packet operation result
+ */
+typedef struct odp_comp_packet_result_t {
+	/** Operation status code */
+	odp_comp_status_t status;
+
+	/** Input packet handle */
+	odp_packet_t pkt_in;
+
+	/** Output packet data range */
+	odp_packet_data_range_t out_data_range;
```

Comment:
To be completely clear: "Specifies offset and length of the data that resulted from a successful de-/compression operation."

So, this is not copy-paste of the range of input params, but where the resulting data is. Maybe it needs to be highlighted that packet data pointer, length or other metadata has not been updated.

Reply:
Updated in the next version.